src/utils/EventWatcher.ts

```diff
@@ -104,6 +97,42 @@ export class EventWatcher {
     });
   }
 
+  getTokenTransfers(logs: Log[], recipient: string) {
+    const ERC20Interface = new utils.Interface(MockERC20Abi.abi);
+    return logs.reduce((logAcc, log) => {
+      try {
+        const parsedLog = ERC20Interface.parseLog(log);
+        if (parsedLog.name === TRANSFER && parsedLog.args.to === recipient) {
+          logAcc.push({
+            token: log.address,
+            amount: parsedLog.args.amount,
+          });
+        }
+        return logAcc;
+      } catch (e) {
+        return logAcc;
+      }
+    }, [] as { token: string; amount: BigNumber }[]);
+  }
+}
+
+export class UniswapXEventWatcher extends EventWatcher<ExclusiveDutchOrderReactor> {
```

Comment:
we should be able to extend this to support X v2?

Reply:
Yes it should be able to, can try in a follow up PR